internal/drivers/lambda/driver.go

```diff
@@ -77,18 +77,28 @@ func (k *driver) Run(ctx context.Context, ref name.Reference) (*drivers.RunResul
 	}
 	clog.FromContext(ctx).Info("Created Lambda function", "name", k.functionName)
 
+	var out *lambda.GetFunctionOutput
+L:
 	for range 10 {
-		out, err := k.client.GetFunction(ctx, &lambda.GetFunctionInput{
+		var err error
+		out, err = k.client.GetFunction(ctx, &lambda.GetFunctionInput{
 			FunctionName: &k.functionName,
 		})
 		if err != nil {
 			return nil, fmt.Errorf("getting Lambda function: %w", err)
 		}
-		if out.Configuration.State == types.StateActive {
-			break
+		switch out.Configuration.State {
```

Comment:
I don't really know enough about the sdk, is there no event's we can "watch" on? similar to the k8s event chan?

Reply:
I couldn't find anything, but I didn't look too hard. In my tests it usually passed or failed within ~20 seconds, so polling doesn't seem too bad. 🤷